### PR TITLE
v3.6.0

### PR DESCRIPTION
**Pull request by x87**
see https://github.com/sannybuilder/dev/issues/10

### CODE REVIEW

**Comment by x87** [comment]
closes https://github.com/sannybuilder/dev/issues/70

File: sa/SASCM.INI [hunk 1]
```diff
@@ -2214,3 +2215,296 @@ DATE=6.14.2013
...
+0EF4=4,clamp_float %1d% min %2d% max %3d% store_to %4d%
+0EF7=4,clamp_int %1d% min %2d% max %3d% store_to %4d%
+0EB3=4,convert_direction_to_quat %1d% dir %2d% %3d% %4d%
+0EF3=4,lerp %1d% %2d% %3d% store_to %4d%
+0E21=1,get_audio_sfx_volume %1d%
+0E22=1,get_audio_radio_volume %1d%
+0E3B=2,get_audiostream_internal %1d% store_to %2d%
+0E28=4,write_struct %1d% offset %2d% size %3d% value %4d%
+0D4E=4,%4d% = read_struct %1d% offset %2d% size %3d% // keep newOpcodes ordering for SB
+0D4F=4,write_struct %1d% offset %2d% size %3d% value %4d% // newOpcodes version of 0E28, the original one doesn't work, this is a fall off
```
Comment (x87) [inline sa/SASCM.INI:2447]: @JuniorDjjr I added CLEO+ definitions after NewOpcodes. 

Can you please explain what is the idea behind having newOpcodes definitions inter-mixed with CLEO+ ones? Can I remove duplicated opcodes?

Comment (JuniorDjjr) [reply sa/SASCM.INI:2447]: It includes the exact same definitions from newOpcodes, so, you can just remove all newOpcodes opcodes (0D**) from CLEO+ definitions without any problem. They are just retrocompatibility.

The opcodes ordering are from source code order (separated by categories), so the opcodes seems very mixed, and during development I removed and added other more useful commands for other categories etc, so don't expect anything organized in terms of numbering.

Comment (x87) [reply sa/SASCM.INI:2447]: Ok, thanks.

@JuniorDjjr Can you clarify one more thing for me? newOpcodes have these definitions:
```
0D02=2,%2d% = matrix %1d% y_angle
0D03=2,%2d% = matrix %1d% x_angle
```
in CLEO+ those defined as
```
0D02=2,%2d% = matrix %1d% x_angle // keep NewOpcodes order
0D03=2,%2d% = matrix %1d% y_angle // keep NewOpcodes order
```

note swapped X and Y in descriptions. Which version is correct?

Comment (JuniorDjjr) [reply sa/SASCM.INI:2447]: This is interesting.
The only place where 0D02=y is in newOpcodes' sascm.ini. In his source code, 0D02=x (considering the name, I'm not sure reading the code)
```
// 0D02=2,%2d% = matrix %1d% x_angle
// 0D02: 1@ = matrix 0@ x_angle
// SCR: GET_MATRIX_X_ANGLE
OpcodeResult WINAPI Matrix::GetXAngle(CScriptThread* thread)
{
	OpcodeParams params(thread, 2);
	RwMatrix *matrix;
	params >> matrix;
	float x = matrix->right.x;
	float y = matrix->right.y;
	float z = matrix->right.z;
	float angle = CGeneral::GetATanOfXY(z, sqrt(x * x + y * y)) * 57.295776f - 90.0f;
	while(angle < 0.0)
		angle += 360.0;
	params << angle;
    return OR_CONTINUE;
}
```
opcodes.txt is `0D02: 1@ = matrix 0@ x_angle`.
So, I'm not 100% sure, but everything indicates that it was a sascm.ini typo.
For Y angle, he uses the same code with `matrix->up`. Someone with more experience in 3D mathematics can confirm this.

**Pull request merged**